packages/react/src/common/is-equal.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Deeply compare two values to determine if they are equal.
+ * Supports primitives and serializable objects.
+ * @param {unknown} value First value to compare
+ * @param {unknown} other Second value to compare
+ * @returns {boolean} True if the values are equal
+ */
+export function isEqual(value: unknown, other: unknown): boolean {
+  if (value === other) {
+    return true;
+  }
+
+  return JSON.stringify(value) === JSON.stringify(other);
```

Comment:
Not sure if we care about this but if the object is not ordered the same order, this will resolve to false:
Could be considered correct, but also wrong here, really not sure.

```js
obj = {}; obj.name="X"; obj.os="linux";
JSON.stringify(obj);
obj = {}; obj.os="linux"; obj.name="X";
JSON.stringify(obj);
```

Reply:
Yeah, I considered that but thought it was unlikely to be an issue. I could either update the check to handle this use or update the JSDocs to mention the limitation. I could also remove this entirely and just do a reference comparison.

Reply:
there's also the option of deep checking the object by traversing through it. Would a reference comparison for an object always return "false" since an updated configuration from the provider would usually end up re-parsing the incoming configuration and creating new objects from it?

Reply:
I've updated the check to traverse the object as @ajwootto suggested.